app/src/main/java/org/p2p/wallet/auth/ui/pin/newcreate/NewCreatePinFragment.kt

```diff
@@ -0,0 +1,92 @@
+package org.p2p.wallet.auth.ui.pin.newcreate
+
+import android.os.Bundle
+import android.view.View
+import androidx.activity.addCallback
+import org.koin.android.ext.android.inject
+import org.p2p.uikit.natives.showSnackbarShort
+import org.p2p.wallet.R
+import org.p2p.wallet.auth.ui.onboarding.OnboardingFragment
+import org.p2p.wallet.auth.ui.pin.select.TouchIdFragment
+import org.p2p.wallet.common.analytics.constants.ScreenNames
+import org.p2p.wallet.common.analytics.interactor.ScreensAnalyticsInteractor
+import org.p2p.wallet.common.mvp.BaseMvpFragment
+import org.p2p.wallet.databinding.FragmentNewCreatePinBinding
+import org.p2p.wallet.utils.popBackStack
+import org.p2p.wallet.utils.popBackStackTo
+import org.p2p.wallet.utils.replaceFragment
+import org.p2p.wallet.utils.vibrate
+import org.p2p.wallet.utils.viewbinding.viewBinding
+
+class NewCreatePinFragment :
+    BaseMvpFragment<NewCreatePinContract.View, NewCreatePinContract.Presenter>(R.layout.fragment_new_create_pin),
+    NewCreatePinContract.View {
+
+    companion object {
+        fun create() = NewCreatePinFragment()
+    }
+
+    override val presenter: NewCreatePinContract.Presenter by inject()
+
+    private val binding: FragmentNewCreatePinBinding by viewBinding()
+    private val analyticsInteractor: ScreensAnalyticsInteractor by inject()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        with(binding) {
+            toolbar.setNavigationOnClickListener { presenter.clearUserData() }
```

Comment:
should we navigate back?

Reply:
`navigateBack()` is called from presenter in that case

Reply:
But clearing user data is not needed here, so I just navigate back